commit 559464ab33e64b5b2f26b8644e01479f75e97120
Author: Lekensteyn <lekensteyn@gmail.com>
Date:   Thu Feb 2 11:41:35 2012 +0100

    Fix crash on invalid PmMethod
    
    The loop indeed stops when there is no pm method available... but it then
    returns -1 which crashes the daemon when bb_pm_method[method_index] is accessed
    the next time.

diff --git a/src/bbconfig.c b/src/bbconfig.c
index 1799ec6..d1ecefb 100644
--- a/src/bbconfig.c
+++ b/src/bbconfig.c
@@ -107,8 +107,8 @@ enum bb_pm_method bb_pm_method_from_string(char *value) {
   /* loop backwards through all possible values. If no valid value is found,
    * assume the first element ("none") */
   enum bb_pm_method method_index = PM_METHODS_COUNT;
-  while (method_index-- > 0) {
-    if (strcmp(value, bb_pm_method_string[method_index]) == 0) {
+  while (method_index > 0) {
+    if (strcmp(value, bb_pm_method_string[--method_index]) == 0) {
       break;
     }
   }
